src/portkey/awssig.clj

```diff
@@ -126,3 +127,75 @@
         headers (assoc headers "Authorization" auth)]
     (assoc req :headers headers)))
 
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Signing a request v2
+;; https://docs.aws.amazon.com/AmazonSimpleDB/latest/DeveloperGuide/HMACAuth.html#REST_RESTAuth
+
+(def x-amz-date-formatter-sign-2
+  (java.time.format.DateTimeFormatter/ofPattern "yyyy-MM-dd'T'HH:mm:ss'Z'"))
+
+(defn- hmac-sha-256-sign2
+   [^bytes secret ^String s]
+   (-> (javax.crypto.Mac/getInstance "HmacSHA256")
+         (doto (.init (javax.crypto.spec.SecretKeySpec. secret "HmacSHA256")))
+         (.doFinal (.getBytes s "UTF8"))))
+
+(defn- utf8-natural-byte-cmp
+   "Sort the UTF-8 query string components by parameter name with natural byte ordering.
+   I wonder if it really matters for the input we deal with"
+   [^String a ^String b]
+   (let [sa a sb b
+         a (.getBytes a "UTF8")
+         b (.getBytes b "UTF8")
+         n (min (alength a) (alength b))]
+        (loop [i 0]
+          (if (< i n)
+            (let [r (- (aget a i) (aget b i))]
+               (if (zero? r)
+                 (recur (inc i))
+                 r))
+            (- (alength a) (alength b))))))
+
+(defn- create-canonicalized-query-string-v2
+   "Step 1 : Create a canonicalized query-string for building the signature v2."
+   [& {:keys [query-params]}]
+   (x/str
+     (comp
+       (x/sort-by key utf8-natural-byte-cmp)
+       (x/for [[k v] %]
+              (str (codec/url-encode k) "=" (codec/url-encode v)))
+       (interpose "&"))
+     query-params))
+
+(defn- create-string-to-sign-v2
+   "Step 2 : Gather request information before signing."
+   [verb host uri qs]
+   (->> [verb "\n" host "\n" uri "\n" qs]
```

Comment:
Maybe just `(str verb "\n" host "\n" uri "\n" qs)`

Reply:
Yes that's a good one 👍